.github/workflows/exercise-7-pipeline.yml

```
@@ -14,3 +14,20 @@ jobs:
     - uses: actions/checkout@v2
     - run: docker build --tag todo-app:test --target test .
     - run: docker run todo-app:test
+
+  deploy:
+    name: Push to Docker Hub and trigger app service deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/exercise-9' }}
+    steps:
+      - uses: actions/checkout@v2
```

I know this is following some out-of-date advice from us, but worth noting that your actions currently have a warning because this one is out of date - we can bump that to v4 to fix it 🙂 
```suggestion
      - uses: actions/checkout@v4
```